src/collection.js

```diff
@@ -72,15 +72,15 @@ export class SyncResultObject {
     if (!Array.isArray(this[type])) {
       return;
     }
+    if (!Array.isArray(entries)) {
+      entries = [entries];
+    }
     // Deduplicate entries by id. If the values don't have `id` attribute, just
     // keep all.
-    const deduplicated = this[type].concat(entries).reduce((acc, cur) => {
-      const existing = acc.filter(
-        r => (cur.id && r.id ? cur.id != r.id : true)
-      );
-      return existing.concat(cur);
-    }, []);
-    this[type] = deduplicated;
+    const ids = new Map(entries.map(({ id }) => [id, true]));
+    this[type] = this[type]
+      .filter(({ id }) => !id || !ids.has(id))
+      .concat(entries);
```

Comment:
What if we see `result.add("skipped", [{id: 1, name: "a"}, {id: 1, name: "b"}]`?